core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -589,6 +589,14 @@ class TeamPolicy
   TeamPolicy(const internal_policy& p) : internal_policy(p) {
     first_arg = false;
   }
+  void check_valid_storage_level_argument(int level) const {
```

Comment:
This should be a free function , not a private member function of a class template. It's not dependent on the class template parameters and there's no reason for us to stamp out a new version of it for every combination of `TeamPolicy` properties.